hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/DefineGroupsLayout.java

```diff
@@ -331,22 +334,29 @@ private void setGroupsValidation(final RolloutGroupsValidation validation) {
         }
 
         // validate the single groups
-        final int maxTargets = quotaManagement.getMaxTargetsPerRolloutGroup();
         final boolean hasRemainingTargetsError = validationStatus == ValidationStatus.INVALID;
+        singleGroupsValidation(validation, lastRow, hasRemainingTargetsError);
+    }
+
+    private void singleGroupsValidation(final RolloutGroupsValidation validation, final GroupRow lastRow,
+            final boolean remainingError) {
+        final int maxTargets = quotaManagement.getMaxTargetsPerRolloutGroup();
+
         for (int i = 0; i < groupRows.size(); ++i) {
             final GroupRow row = groupRows.get(i);
             // do not mask the 'remaining targets' error
-            if (hasRemainingTargetsError && row.equals(lastRow)) {
+            if (remainingError && row.equals(lastRow)) {
```

Comment:
You don't need the _remainingError_ method parameter, and _ hasRemainingTargetsError_ field, just use the `validationStatus == ValidationStatus.INVALID` here directly:
```suggestion
            if (validationStatus == ValidationStatus.INVALID && row.equals(lastRow)) {
```